docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -592,6 +646,18 @@ determines the name administrators will ultimately use to deploy the charm. E.g.
         :start-at: name:
         :end-before: parts:
 
+.. admonition:: Best Practice
+    :class: hint
+
+    The name should be slug-oriented (ASCII lowercase letters, numbers, and
+    hyphens) and follow the pattern ``<workload name in full>[<function>][-k8s]``.
+    For example, ``argo-server-k8s``.
+
+    Don't include an organisation or publisher in the name.
+
+    Don't add an ``operator`` or ``charm`` prefix or suffix. For naming a
+    repository, see :ref:`initialise-a-charm`.
+
```

Comment:
Thank you for including the negatives (which I forgot to include when I added this text originally).